beamer/config/commands.py

```diff
@@ -110,19 +110,20 @@ def read(
 
     request_manager = deployment.obtain_contract(w3, "chain", "RequestManager")
     fill_manager = deployment.obtain_contract(w3, "chain", "FillManager")
-    start_block = min(
-        deployment.chain.contracts["RequestManager"].deployment_block,
-        deployment.chain.contracts["FillManager"].deployment_block,
-    )
-    fetcher = EventFetcher(
-        w3, (request_manager, fill_manager), start_block=start_block, confirmation_blocks=0
-    )
 
     if state_path.exists():
         config = Configuration.from_file(state_path)
+        start_block = config.block
     else:
+        start_block = min(
+            deployment.chain.contracts["RequestManager"].deployment_block,
+            deployment.chain.contracts["FillManager"].deployment_block,
+        )
         config = Configuration.initial(chain_id, start_block)
 
+    fetcher = EventFetcher(
+        w3, (request_manager, fill_manager), start_block=start_block, confirmation_blocks=0
```

Comment:
Are we potentially replaying the events for `start_block` twice? Meaning, is `config.block` and `EventFetcher.start_block` both inclusive? 
I think it is most likely not a problem but it would be great to get this straight.

Reply:
Good point! Changed.